src/ai/ai.cpp

```diff
@@ -1833,11 +1887,23 @@ bool will_accept_crisis_peace_offer(sys::state& state, dcon::nation_id to, dcon:
 
 void update_war_intervention(sys::state& state) {
 	for(auto& gp : state.great_nations) {
-		if(state.world.nation_get_is_player_controlled(gp.nation) == false && state.world.nation_get_is_at_war(gp.nation) == false) {
+		if(state.world.nation_get_is_player_controlled(gp.nation) == false){
 			bool as_attacker = false;
 			dcon::war_id intervention_target;
 			[&]() {
 				for(auto w : state.world.in_war) {
+					//GPs will try to intervene in wars to protect smaller nations in the same cultural union
+					if(command::can_intervene_in_war(state, gp.nation, w, false)) {
+						for(auto par : w.get_war_participant()) {
```

Comment:
instead of looping over everyone in the war, just check the primary defender -- this will prevent the US jumping into any war that texas gets involved in as a defensive ally, for an example of the weird things that could happen